dfs/restore.go

```diff
@@ -18,89 +18,82 @@ import (
 	"encoding/json"
 	"errors"
 	"io"
+	"os/exec"
 	"strings"
+	"sync"
 
 	"github.com/control-center/serviced/volume"
 	"github.com/zenoss/glog"
 )
 
-var ErrRestoreNoInfo = errors.New("backup is missing metadata")
+var (
+	ErrRestoreNoInfo        = errors.New("backup is missing metadata")
+	ErrInvalidBackupVersion = errors.New("backup has an invalid version")
+)
+
+// ExtractBackupInfo extracts the backup metadata from a tarball on disk in as
+// cheaply a manner as possible. The serialized BackupInfo is stored at the
+// front of the tarball to facilitate this.
+func ExtractBackupInfo(filename string) (*BackupInfo, error) {
```

Comment:
I don't think this makes sense to be in this file.  I think this should probably go into backupinfo.go